clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractFetch.java

```diff
@@ -92,6 +88,23 @@ public AbstractFetch(final LogContext logContext,
         this.time = time;
     }
 
+    /**
+     * Check if the node is disconnected and unavailable for immediate reconnection (i.e. if it is in
+     * reconnect backoff window following the disconnect).
+     *
+     * @param node {@link Node} to check for availability
+     * @see NetworkClientUtils#isUnavailable(KafkaClient, Node, Time)
+     */
+    protected abstract boolean isUnavailable(Node node);
```

Comment:
I am actually not sure if we need this as NetworkClientDelegate is already doing the connection checking.  If the node is unavailable for reconnection, then it would fail the unsentRequest and presumably trigger a retry.

The original code does the connection checking because the request is sent right after its creation; however, there's a time gap between the creation and the actual network IO.

I think it is harmless to leave it here but it would be great if we could clean them up later.

Reply:
@philipnee `AbstractFetch` is used by both the current `Fetcher` as well as the `FetchRequestManager`. I'd have to refactor the code to rid ourselves of it.

What do you think about filing a new Jira to track a follow-up change for this?

Reply:
I have filed KAFKA-15556 (_Remove NetworkClientDelegate methods isUnavailable, maybeThrowAuthFailure, and tryConnect_) to address this issue since it's affecting other `RequestManager` implementations.